sysl2/sysl/swagger/import.go

```diff
@@ -0,0 +1,319 @@
+		globalParams: globalParams,
+	}
+	if err := si.Write(); err != nil {
+		return "", err
+	}
+
+	return result.String(), nil
+}
+
+type swaggerImporter struct {
+	io.Writer
+	ind     *IndentWriter
+	swagger *spec.Swagger
+	args    OutputData
+
+	types        TypeList
+	endpoints    map[string][]Endpoint
+	globalParams Parameters
+}
+
+func (si *swaggerImporter) mustWrite(s string) {
+	if _, err := si.Writer.Write([]byte(s)); err != nil {
+		panic(err)
+	}
+}
+
+const PushIndent = "&& >>"
+const PopIndent = "&& <<"
+const BlankLine = "&& !!"
+
+func (si *swaggerImporter) writeLines(lines ...string) {
+	for _, l := range lines {
+		switch l {
+		case PushIndent:
+			si.ind.Push()
+		case PopIndent:
+			si.ind.Pop()
+		case BlankLine:
+			si.mustWrite("\n")
+		default:
+			_ = si.ind.Write() // nolint: errcheck
+			si.mustWrite(l + "\n")
+		}
+	}
+}
+
+func (si *swaggerImporter) Write() error {
+
+	if err := si.writeHeader(); err != nil {
+		return err
+	}
+
+	if si.swagger.Info != nil {
+		if err := si.writeInfo(); err != nil {
+			return err
+		}
+	}
+
+	if si.swagger.BasePath != "" {
+		si.writeLines(fmt.Sprintf("%s:", si.swagger.BasePath), PushIndent, BlankLine)
+	}
+	for _, method := range methodDisplayOrder {
+		for _, endpoint := range si.endpoints[method] {
+			si.writeEndpoint(method, endpoint)
+			si.writeLines(BlankLine)
+		}
+	}
+	if si.swagger.BasePath != "" {
+		si.writeLines(PopIndent)
+	}
+
+	si.writeLines("#" + strings.Repeat("-", 75))
+	si.writeLines("# definitions")
+	for _, t := range si.types {
+		si.writeLines(BlankLine)
+		si.writeDefinition(t)
+	}
+
+	return nil
+}
+
+func (si *swaggerImporter) writeInfo() error {
+
+	desc := ""
+	if info := si.swagger.Info; info != nil {
+		desc = getDescription(info.Description)
+		values := []string{
+			"version", info.Version,
+			"host", si.swagger.Host,
+			"license", "",
+			"termsOfService", info.TermsOfService}
+		for i := 0; i < len(values); i += 2 {
+			key := values[i]
+			val := values[i+1]
+			if val != "" {
+				si.writeLines(fmt.Sprintf("@%s = %s", key, quote(val)))
+			}
+		}
+	}
+	si.writeLines("@description =:", PushIndent, "| "+getDescription(desc), PopIndent, BlankLine)
+	return nil
+}
+
+func getDescription(d string) string {
+	if d == "" {
+		return "No description."
+	}
+	return d
+}
+
+func quote(s string) string {
+	return `"` + s + `"`
+}
+
+func (si *swaggerImporter) writeHeader() error {
+
+	si.mustWrite(`##########################################
+##                                      ##
+##  AUTOGENERATED CODE -- DO NOT EDIT!  ##
+##                                      ##
+##########################################
+
+`)
+	title := ""
+	if si.swagger.Info != nil {
+		title = si.swagger.Info.Title
+	}
+
+	si.writeLines(fmt.Sprintf("%s %s [package=%s]:", si.args.AppName, quote(title), quote(si.args.Package)))
+	si.ind.Push()
+
+	return nil
+}
+
+func buildQueryString(params []Param) string {
+	query := ""
+	if len(params) > 0 {
+		parts := []string{}
```

Comment:
nit: nil slice is preferred.
`var parts []string`

Reply:
fixed all 3